bin/bossSwitch.py

```diff
@@ -0,0 +1,210 @@
+
+        elif args.action == "off":
+            print("Turning the BossDB off...")
+            stopInstances()
+
+#Executed actions
+def startInstances():
+    """
+        Method used to start necessary instances
+    """
+    #Use auto scaling groups last saved configuration
+    ASGdescription = load_obj('ASGdescriptions')
+    activitiesD = [ASGdescription["AutoScalingGroups"][0]["MinSize"], ASGdescription["AutoScalingGroups"][0]["MaxSize"],ASGdescription["AutoScalingGroups"][0]["DesiredCapacity"]]
+    endpointD = [ASGdescription["AutoScalingGroups"][1]["MinSize"], ASGdescription["AutoScalingGroups"][1]["MaxSize"],ASGdescription["AutoScalingGroups"][1]["DesiredCapacity"]]
+    vaultD = [ASGdescription["AutoScalingGroups"][2]["MinSize"], ASGdescription["AutoScalingGroups"][2]["MaxSize"],ASGdescription["AutoScalingGroups"][2]["DesiredCapacity"]]
+   
+    #Start vault instance
+    print("Starting vault...")
+    client.update_auto_scaling_group(AutoScalingGroupName=vaultg, MinSize = vaultD[0] , MaxSize =  vaultD[1], DesiredCapacity = vaultD[2])
+    client.resume_processes(AutoScalingGroupName=vaultg,ScalingProcesses=['HealthCheck'])
+    time.sleep(120)
+    print("Vault instance running")
+
+    #Import vault content:
+    subprocess.call('./bastion.py ' + args.vpc + ' vault-unseal',shell=True)
+    print("Importing vault content")
+    subprocess.call('./bastion.py ' + args.vpc + ' vault-import < ../config/vault_export.json', shell=True)
+    print(bcolors.WARNING + "Successful import" + bcolors.ENDC)    
+
+    #Start endpoint and activities instances
+    print("Starting endpoint, and activities...")
+    client.update_auto_scaling_group(AutoScalingGroupName=endpoint, MinSize = endpointD[0] , MaxSize = endpointD[1] , DesiredCapacity = endpointD[2])
+    client.resume_processes(AutoScalingGroupName=endpoint,ScalingProcesses=['HealthCheck'])
+    
+    client.update_auto_scaling_group(AutoScalingGroupName=activities, MinSize = activitiesD[0] , MaxSize = activitiesD[1] , DesiredCapacity = activitiesD[2])
+    client.resume_processes(AutoScalingGroupName=activities,ScalingProcesses=['HealthCheck'])
+
+    print(bcolors.OKGREEN + "TheBoss is on" + bcolors.ENDC)
+
+
+def stopInstances():
+    """
+        Method used to stop currently running instances
+    """
+    #Save current ASG descriptions:
+    print("Saving current auto scaling group configuration...")
+    response = client.describe_auto_scaling_groups(AutoScalingGroupNames=[endpoint,activities,vaultg])
+    save_obj(response,'ASGdescriptions')
+    
+    #Export vault content:
+    print("Exporting vault content...")
+    subprocess.call('./bastion.py ' + args.vpc + ' vault-export > ../config/vault_export.json', shell=True)
+    print(bcolors.WARNING + "Successful export" + bcolors.ENDC)
+
+    #Switch off:
+    print("Stopping all Instances...")
+    client.update_auto_scaling_group(AutoScalingGroupName=endpoint, MinSize = 0 , MaxSize = 0 , DesiredCapacity = 0)
```

Comment:
How come the example file include `consul` and `auth` ASG but they are not included in the stopping / starting? Like I commented with the starting section I would loop over all of the entries in the ASG config file. That would make this script more future proof.

For shutting down it seems like there are no special instructions needed (like with starting Vault and unsealing and importing data) so you could just use the order of ASGs in the config file.

Reply:
To load a json file in the same order it was specified: https://stackoverflow.com/questions/6921699/can-i-get-json-to-load-into-an-ordereddict